hadoop-ozone/dist/src/main/compose/upgrade/testlib.sh

```diff
@@ -94,8 +94,8 @@ run_test() {
     RESULT=1
   fi
 
-  generate_report 'upgrade' "$RESULT_DIR"
   copy_results "$test_subdir" "$ALL_RESULT_DIR"
+  generate_report 'upgrade' "$RESULT_DIR"
```

Comment:
Can we remove the `generate_report` call completely?

`run_test` is called from `test.sh`, where it is immediately followed by `generate_report`:

https://github.com/apache/ozone/blob/c87b059b0f475c8af08f1ae7ead9f5e193e9df57/hadoop-ozone/dist/src/main/compose/upgrade/test.sh#L35-L37

Reply:
I guess so. @errose28

Reply:
I think that is indeed redundant at the moment. But I recall @errose28 mentioning in the future we would need to have a matrix to run different set of versions. Something like:

```
run_test non-rolling-upgrade 1.1.0 1.2.0
run_test non-rolling-upgrade 1.1.0 1.3.0
run_test non-rolling-upgrade 1.2.0 1.3.0
```

In this case we might still need the `generate_report` in `run_test`, depending on where the XML files are written to. But it looks to me `copy_results` alone is enough.